### ISSUE TITLE
What do the output letters signify?

### ISSUE THREAD

**Issue by iboates** [lucidrains/stylegan2-pytorch#26]
What do the letters "G", "D", "GP" and "PL" in the output signify?

```
0% 0/90000 [00:00<?, ?it/s]G: 1.51 | D: 0.21 | GP: 0.00 | PL: 0.00
0% 49/90000 [02:46<81:43:57,  3.27s/it]G: 2.06 | D: 0.05 | GP: 0.50 | PL: 0.02
0% 97/90000 [05:04<71:55:48,  2.88s/it]G: 1.63 | D: 0.00 | GP: 0.00 | PL: 0.03
0% 149/90000 [07:35<72:52:45,  2.92s/it]G: 1.04 | D: 0.12 | GP: 0.00 | PL: 0.03
```

**Comment by lucidrains** [comment]
@iboates you are back! any interesting training results? :)

Those are the vital signs of the training, the numbers that the network is forced to try to minimize. 

G: generator loss
D: discriminator loss
GP: gradient penalty loss
PL: path length regularization loss

**Comment by iboates** [comment]
The results so far have been disappointing unfortunately. I have been trying to train it on images of maps coming from openstreetmap.

It was making good progress, but after about 70k iterations the model collapsed and began outputting random smears of colour. I was thinking that maybe it had to do with the fact that the data pool was pretty small, and what images I did have had quite a bit of variety.

So I have come up with a way to get much more data, and isolated the maps so that they are always featuring villages or small towns. I'm training again with about 3k of these images, but I can generate many many more. But right now I think I am hitting the upper end of Google Colab, it takes about 7 hours to do 10k iterations, and that is when the checkpoint is made. Since Colab disconnects after 10 hours, I can't really squeeze more data in. I think I have to buy some cloud processing time or something.

**Comment by lucidrains** [comment]
@iboates ahh 3k training set is nothing! you need up to 10k or more!

**Comment by iboates** [comment]
Thanks for the clarification. We're a bit off-topic from the original question, but how many faces did you train it on for thispersondoesnotexist.com? And do all the images have to be the same size or can they just be of similar size & resolution to depict what they are showing?

**Comment by lucidrains** [comment]
@iboates to get to that level of quality, I would recommend the official repository, as it is more optimized. That model was trained on 70k high quality images by Nvidia

**Comment by iboates** [comment]
Do all the images need to be the exact same size?

**Comment by lucidrains** [comment]
@iboates ideally yes!
